Docs for license autoloading. (#11886)
hashicorp/vault: website/content/docs/enterprise/license/autoloading.mdx

---
layout: docs
page_title: License Autoloading
description: An overview of license autoloading.
---

Prior to Vault 1.8, Vault Enterprise would be licensed using special binaries
that contained embedded licenses, or via a license written into Vault storage

> using the `PUT sys/license` API.

Reviewer: Is it worth linking to said API here?

Author: I was going to say no, but it was taking longer to justify than it would to just make the link.

As of Vault 1.8 those options still exist but are deprecated, and the recommended
mechanism for managing licenses is called License Autoloading.  New clusters are
required to use the autoloading method.

License Autoloading can be done using one of these mechanisms, in increasing order
of priority (i.e. `VAULT_LICENSE` takes precedence over `VAULT_LICENSE_PATH`, which
takes precedence over `license_path` in config.)
* [license_path in config](/docs/configuration#license_path)
* [VAULT_LICENSE_PATH environment variable](/docs/commands#vault_license_path)
* [VAULT_LICENSE environment variable](/docs/commands#vault_license)

Each node needs to be licensed, they should all use the same license string, and
they should all use autoloading or none should. Warnings will be written to

> the log when there are discrepancies.

Reviewer: The first sentence in this paragraph reads a little awkwardly to me. Here is my suggested replacement:

Nodes within a cluster should be consistently licensed. All nodes in a cluster should use autoloading, or none should. If autoloading is used, all nodes should use the same license string.

Author: Better, thanks.

> If autoloading is used any existing stored license will be ignored.  The presence

Reviewer: ```suggestion
If autoloading is used, any existing stored license will be ignored.  The presence
```

> of a stored license in conjunction with autoloaded license will also result in

Reviewer: ```suggestion
of a stored license in conjunction with an autoloaded license will also result in
```

> logged warnings.  Having migrated to autoloading, it is recommended to use

Reviewer: ```suggestion
logged warnings.  Once a migration to autoloading is completed, it is recommended to use
```
